src/rgw/rgw_admin.cc

```diff
@@ -6444,10 +6447,28 @@ int main(int argc, const char **argv)
     }
 
     rgw_data_sync_status sync_status;
-    ret = sync.read_sync_status(&sync_status);
-    if (ret < 0 && ret != -ENOENT) {
-      cerr << "ERROR: sync.read_sync_status() returned ret=" << ret << std::endl;
-      return -ret;
+    if (specified_shard_id) {
+      set<string> lagging_buckets;
+      set<string> recovering_buckets;
+      rgw_data_sync_marker sync_marker;
+      ret = sync.read_shard_status(shard_id, lagging_buckets, recovering_buckets, &sync_marker, 
+                                   max_entries_specified ? max_entries : 1000);
```

Comment:
i think a default of 10 or 20 entries is more reasonable here - any more than that will overwhelm the user, who will likely want to follow up on these entries with a `bucket sync status` command